File: CSharp/cseries/cseries.cs
using System.Runtime.InteropServices;

[StructLayout(LayoutKind.Sequential)]
public struct real_range {
    public float min;
    public float max;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_color {
    public float red;
    public float green;
    public float blue;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_color_alpha {
    public float alpha;
    public float red;
    public float green;
    public float blue;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_vector2d {
    public float x;
    public float y;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_rotation2d {
    public float yaw;
    public float pitch;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_vector3d {
    public float x;
    public float y;
    public float z;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_offset3d {
    // X-Component
    public float i;
    // Y-Component
    public float j;
    // Z-Component
    public float k;
}
[StructLayout(LayoutKind.Sequential)]
public struct real_quaternion {
    // X-Component
    public float i;
    // Y-Component
    public float j;
    // Z-Component
    public float k;
    // ?-Component
    public float w;
}
[StructLayout(LayoutKind.Sequential)]
public struct byte_color_alpha {
    public byte alpha;
    public byte red;
    public byte green;
    public byte blue;
}
[StructLayout(LayoutKind.Sequential)]
public struct short_range {
    public short min;
    public short max;
}
[StructLayout(LayoutKind.Sequential)]
public struct short_point {
    public short x;
    public short y;
}
[StructLayout(LayoutKind.Explicit, Size=4)]
public struct s_ident {
    [FieldOffset(0)]
    public int Tag;
    [FieldOffset(0)]
    public short index;
    [FieldOffset(2)]
    public short salt;
    public s_ident(int tag = -1) {
        salt = (short)tag;
        index = (short)tag;
        Tag = tag;
    }
    s_ident(ref s_ident ident) {
        index = ident.index;
        salt = ident.salt;
        Tag = ident.Tag;
    }
}

File: VisualBasic/cseries/cseries.vb
Imports System.Runtime.InteropServices

' tag_name_length = uint;

' tag_name_reference = UnmanagedType.LPStr;

<StructLayout(LayoutKind.Sequential)>
Public Structure real_range
    Public min As Single
    Public max As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_color
    Public red As Single
    Public green As Single
    Public blue As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_color_alpha
    Public alpha As Single
    Public red As Single
    Public green As Single
    Public blue As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_vector2d
    Public x As Single
    Public y As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_rotation2d
    Public yaw As Single
    Public pitch As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_vector3d
    Public x As Single
    Public y As Single
    Public z As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_offset3d
    ' X-Component
    Public i As Single
    ' Y-Component
    Public j As Single
    ' Z-Component
    Public k As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure real_quaternion
    ' X-Component
    Public i As Single
    ' Y-Component
    Public j As Single
    ' Z-Component
    Public k As Single
    ' ?-Component
    Public w As Single
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure byte_color_alpha
    Public alpha As Byte
    Public red As Byte
    Public green As Byte
    Public blue As Byte
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure short_range
    Public min As Short
    Public max As Short
End Structure
<StructLayout(LayoutKind.Sequential)>
Public Structure short_point
    Public x As Short
    Public y As Short
End Structure
<StructLayout(LayoutKind.Explicit, Size:=4)>
Public Structure s_ident
    <FieldOffset(0)>
    Public Tag As Integer
    <FieldOffset(0)>
    Public index As Short
    <FieldOffset(2)>
    Public salt As Short
    Public Sub New(Optional tag__1 As Integer = -1)
        salt = CShort(tag__1)
        index = CShort(tag__1)
        Tag = tag__1
    End Sub
    Private Sub New(ByRef ident As s_ident)
        index = ident.index
        salt = ident.salt
        Tag = ident.Tag
    End Sub
End Structure
